modules/nf-core/fqtk/main.nf

```diff
@@ -0,0 +1,47 @@
+process FQTK {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda "bioconda::fqtk=0.2.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/fqtk:0.2.0--h9f5acd7_0' :
+        'quay.io/biocontainers/fqtk:0.2.0--h9f5acd7_0' }"
+
+    input:
+    tuple val(meta), path(sample_sheet), path(fastq_dir)
+    val(fastq_readstructure_pairs)
+
+    output:
+    tuple val(meta), path('output/*R*.fq.gz')                       , emit: sample_fastq
```

Comment:
```suggestion
    // if the `--output-types` argument is changed, this needs to change
    tuple val(meta), path('output/*R*.fq.gz')                       , emit: sample_fastq
```

Reply:
@nh13 in the demultiplex pipeline fqtk will be called by specifying `--demultiplexer fqtk` and proving an extra param `read_structure_manifest.csv` which maps file names to read structure. Then in `demultiplex.nf` the additional input `fastq_readstructure_pair` will be made prior to running `FQTK`. 

I am happy to make any changes, but I think it is easiest to understand this by looking at the current implementation of `fqtk` in `demultiplex` [here lines 147 - 157 ](https://github.com/fulcrumgenomics/nf-core-demultiplex/blob/fqtk_dev/workflows/demultiplex.nf) (currently the read structure manifest is read in from github - this will need to be a param)

Reply:
I think the read structures don't need to be in its own separate file, that's confusing.  I'd prefer them to be a parameter on the command line (or config, though nextfow is supposed to be zero config(TM)!).

Reply:
I understand the desire to reduce the config files, but the read structures need to be paired to the fastq file names and I thought providing a csv would be easier than making two new cl params - one for fastq file names and another for read structures where the list order matters. Thoughts?

Reply:
Good question. The command line has the same issue then, and it’s still error prone. Let’s not have another file. Folks wanting to use this module would have to create a file themselves or programatically. I’ll cash in some credits (if I have any) for no file! Let’s discuss with the other authors if you’re unsure still, I could be wrong.

Reply:
@geoffjentry gave a vote for a cl option, I have updated the workflow and it no longer requires an additional file :)